server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -761,6 +770,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 DiffableUtils.StringSetValueSerializer.getInstance());
             rolloverInfos = DiffableUtils.readImmutableOpenMapDiff(in, DiffableUtils.getStringKeySerializer(), RolloverInfo::new,
                 RolloverInfo::readDiffFrom);
+            if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
I've found that using a constant (e.g. `SYSTEM_INDEX_FLAG_ADDED=V_8_0_0`) makes serialization logic easier to follow when reading, and also easier to change on a backport. Especially so when that version is used in several streaming methods, as is the case here.